docker.Makefile

```diff
@@ -10,7 +10,11 @@ LINTER_IMAGE_NAME = docker-cli-lint$(IMAGE_TAG)
 CROSS_IMAGE_NAME = docker-cli-cross$(IMAGE_TAG)
 VALIDATE_IMAGE_NAME = docker-cli-shell-validate$(IMAGE_TAG)
 E2E_IMAGE_NAME = docker-cli-e2e$(IMAGE_TAG)
+GO_BUILD_CACHE ?= y
 MOUNTS = -v "$(CURDIR)":/go/src/github.com/docker/cli
+ifeq ($(GO_BUILD_CACHE),y)
+MOUNTS += -v "docker-cli-dev-cache:/root/.cache/go-build"
```

Comment:
If you still want `make -f docker.Makefile clean`, that target could do a `docker volume rm docker-cli-dev-cache`